microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/JWTSecurityAnnotationsTest.java

```diff
@@ -0,0 +1,184 @@
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.router.rest.LocalServiceRouterInfoResource;
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.router.rest.routed.RouterDistrictsResource;
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.router.services.DistrictServiceClient;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Test cases for usage of JWT related security MP OpenAPI annotations
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({ JWTSecurityAnnotationsTest.OpenApiExtensionSetup.class })
+@RunAsClient
+public class JWTSecurityAnnotationsTest {
+
+    private final static String PROVIDER_DEPLOYMENT_NAME = "serviceProviderDeployment";
+    private final static String ROUTER_DEPLOYMENT_NAME = "localServicesRouterDeployment";
+
+    private static OnlineManagementClient onlineManagementClient;
+
+    @Deployment(name = PROVIDER_DEPLOYMENT_NAME, order = 1, testable = false)
+    public static Archive<?> serviceProviderDeployment() {
+        //  MP OpenAPI disabled on this Services Provider deployment for testing purposes, we don't need it here
+        String mpConfigProperties = "mp.openapi.extensions.enabled=false";
+        WebArchive deployment = ShrinkWrap.create(
+                WebArchive.class, PROVIDER_DEPLOYMENT_NAME + ".war")
+                .addClasses(ProviderApplication.class)
+                .addClasses(
+                        District.class,
+                        DistrictEntity.class,
+                        DistrictService.class,
+                        InMemoryDistrictService.class,
+                        DistrictsResource.class,
+                        RoutingServiceConstants.class)
+                .addAsManifestResource(new StringAsset(mpConfigProperties), "microprofile-config.properties")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return deployment;
+    }
+
+    @Deployment(name = ROUTER_DEPLOYMENT_NAME, order = 2, testable = false)
+    public static Archive<?> localServicesRouterDeployment() {
+        //  OpenAPI doc here is exposing doc generated only from Local Service Provider "non-routed" JAX-RS endpoints
+        //  as MP Config is used to tell MP OpenAPI to skip doc generation for exposed "routed" JAX-RS endpoints
+        //  Here we also add config properties to reach the Services Provider.
+        String mpConfigProperties = "mp.openapi.scan.exclude.packages=org.jboss.eap.qe.microprofile.openapi.apps.routing.router.rest.routed"
```

Comment:
Can you move the content to a resource file?

Reply:
Same as above: my proposal is to add JavaDoc to the deployment method.